### PR TITLE
Adjust TCP keepalive config

### PR DESCRIPTION
**Pull request by Lanius-collaris**

### CODE REVIEW

File: intra/protect/xdial.go [hunk 1]
```diff
@@ -201,3 +210,24 @@ func (d *RDial) AcceptTCP(network string, local string) (*net.TCPListener, error
 		return nil, errNoTCPMux
 	}
 }
+
+func setKeepAliveConfig(tc *net.TCPConn, keepIdle, keepCnt, keepIntvl int){
```
Comment (ignoramous) [inline intra/protect/xdial.go]: minor: could move this fn and other constants to a new file `intra/protect/sockopt.go`?

File: intra/protect/xdial.go [hunk 2]
```diff
@@ -149,6 +156,8 @@ func (d *RDial) DialTCP(network string, laddr, raddr *net.TCPAddr) (*net.TCPConn
 		return nil, err
 	} else if tc, ok := c.(*net.TCPConn); ok {
 		// d.Dialer.LocalAddr = nil
+		//adjust keepalive config to save battery
+		setKeepAliveConfig(tc, defaultKeepIdle, defaultKeepCnt, defaultKeepIntvl)
```
Comment (ignoramous) [inline intra/protect/xdial.go]: Another way to do so would be to create a `intra/protect/protect.go:MakeNsDialerExt` (similar to `intra/protect/protect.go:MakeNsListenConfigExt`) to pass `Control` fns to the underlying dialer from `ipn/Base.go` and `ipn/Exit.go` (that is, the sock opts should be set by the proxy dialers and not commonly by this "adapter" `xdial.go`).

https://github.com/celzero/firestack/blob/7ea0539e57ba513065dcde893084f70cfedd74b9/intra/protect/protect.go#L176

https://github.com/celzero/firestack/blob/7ea0539e57ba513065dcde893084f70cfedd74b9/intra/ipn/base.go#L42

https://github.com/celzero/firestack/blob/7ea0539e57ba513065dcde893084f70cfedd74b9/intra/ipn/exit.go#L31

**Review by ignoramous — APPROVED** [review]
There's a few nagging linter complaints about `nil` ptr.

File: intra/protect/xdial.go (hunk 2 above)
Comment (Lanius-collaris) [reply intra/protect/xdial.go]: 🙃Go still overrides these options after calling `Control()`.
[pcap.zip](https://github.com/user-attachments/files/16452724/pcap.zip)

**Pull request merged**